src/eliottree/tree_format/_text.py

```diff
@@ -100,14 +100,32 @@ def _format_tree(node, format_node, get_children, options, prefix=u'', depth=0):
                                    depth=depth + 1):
             yield result
 
+def chunked_format_tree(node, format_node, get_children, options=None, chunk_size=0)->Generator[str]:
+    lines = itertools.chain(
+        [format_node(node)],
+        _format_tree(node, format_node, get_children, options or Options()),
+        [u""],
+    )
+    if chunk_size is None or chunk_size<=0:
+        yield u"\n".join(lines)
+    else:
+        write_lines = u"\n".join((line for _, line in zip(range(chunk_size), lines)))
```

Comment:
I think what you want here is [`partition`](https://toolz.readthedocs.io/en/latest/api.html#toolz.itertoolz.partition) from the `toolz` package (which is already a dependency):

```python
from toolz import partition

write_lines = '\n'.join(partition(lines, chunk_size))
```